examples/crud_on_store/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+//! Actual schema is defined [here](https://github.com/primait/event_sourcing.rs/blob/master/src/esrs/postgres/statement.rs#L58)
+use std::convert::TryInto;
+
+use chrono::{DateTime, Utc};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use serde_json::Value;
+use sqlx::types::Json;
+use sqlx::{Executor, Postgres};
+use uuid::Uuid;
+
+use esrs::StoreEvent;
+
+// Get all events by aggregate id: use PgStore<Manager>::by_aggregate_id function
+
+// Delete all events by aggregate id: use PgStore<Manager>::delete function
+
+// Insert event: use PgStore<Manager>::save_event function
+
+// Insert event with given event id. The table should be the aggregate name, not the aggregate_events
+pub async fn insert_event_with_given_event_id<T: Serialize + Sync>(
+    table: &str,
+    id: Uuid,
+    aggregate_id: Uuid,
+    payload: &T,
+    occurred_on: DateTime<Utc>,
+    sequence_number: i32,
+    executor: impl Executor<'_, Database = Postgres>,
+) {
+    // The path in the include_str! is the one for the given query. In your codebase you can copy the
+    // file or directly use the content
+    let name: String = format!("{}_events", table);
+    let query: String = format!(include_str!("../../../src/esrs/postgres/statements/insert.sql"), name);
+    let _ = sqlx::query(query.as_str())
+        .bind(id)
+        .bind(aggregate_id)
+        .bind(Json(&payload))
+        .bind(occurred_on)
+        .bind(sequence_number)
+        .execute(executor)
+        .await
+        .expect("Failed to insert event");
+}
+
+// Get event by event id. The table should be the aggregate name, not the aggregate_events
+pub async fn get_event_by_id<T: DeserializeOwned>(
+    table: &str,
+    event_id: Uuid,
+    executor: impl Executor<'_, Database = Postgres>,
+) -> Option<StoreEvent<T>> {
+    let name: String = format!("{}_events", table);
+    let query: String = format!(include_str!("../statements/select_by_event_id.sql"), name);
+
+    sqlx::query_as::<_, Event>(query.as_str())
+        .bind(event_id)
+        .fetch_one(executor)
+        .await
+        .ok()
+        .map(|v| v.try_into().expect("Failed to cast database event into T"))
+}
+
+// Update event by event id
+pub async fn update_event_by_event_id<T: Serialize + Sync>(
```

Comment:
Should we recommend people change the "occurred_on" here too?

Reply:
I think that yes, we can do it. But i think that probably if someone wants to update the event store table wants to update the payload; like adding a field that was missing, or to fix a wrong value or whatever. I don't know in which case someone could want to edit the occurred_on field

Reply:
I mean more is it a good idea if you're updating an event payload to /also/ update the occurred on, since now the values in the event are not the original ones? I'm not sure either way

Reply:
I don't think so because the order of the event is important being that lot of query order by occurred_on (yes, and by sequence number too, but firstly it orders by occurred_on. The sequence number ordering is only because we don't want that two events that happen in the same millisecond come up in the wrong order)

Reply:
Makes sense!